processing/src/main/java/io/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -231,6 +263,45 @@ private IncrementalIndex makeIncrementalIndex(GroupByQuery query, Sequence<Row>
   }
 
   @Override
+  public Function<Row, Row> makePostComputeManipulatorFn(final GroupByQuery query, final MetricManipulationFn fn)
+  {
+    final Set<String> postNames = Sets.newHashSet(
+        Collections2.transform(
+            query.getPostAggregatorSpecs(),
+            new Function<PostAggregator, String>()
+            {
+              @Override
+              public String apply(PostAggregator input)
+              {
+                return input.getName();
+              }
+            }
+        )
+    );
+    return new Function<Row, Row>()
+    {
+      @Nullable
+      @Override
+      public Row apply(Row input)
+      {
+        if (!(input instanceof MapBasedRow)) {
```

Comment:
Good question, the default behavior is to use the preComputeManipulatorFn, which also only computes for MapBasedRow. I couldn't tell you why that is. This function is a copy of the pre function